wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/BuildConfigFileElement.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema.internal.parser;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.squareup.wire.schema.Location;
+
+/**
+ * A single {@code build.wire} file. This file is structured similarly to a {@code .proto} file, but
+ * with different elements.
+ *
+ * <h3>File Structure</h3>
+ *
+ * A project may have 0 or more {@code build.wire} files. These files should be in the same
+ * directory as the {@code .proto} files so they may be automatically discovered by Wire.
+ *
+ * <p>Each file starts with a syntax declaration. The syntax must be "wire2". This is followed by an
+ * optional package declaration, which should match to the package declarations of the {@code
+ * .proto} files in the directory.
+ *
+ * <p>Build configs may import any number of proto files. Note that it is an error to import {@code
+ * build.wire} files. These imports are used to resolve types specified later in the file.
+ *
+ * <p>Build configs may specify any number of type configurations. These specify a fully qualified
+ * type, its target Java type, and an adapter to do the encoding and decoding.
+ *
+ * <pre>   {@code
+ *
+ *   syntax = "wire2";
+ *   package squareup.dinosaurs;
+ *
+ *   import "squareup/geology/period.proto";
+ *
+ *   // Roar!
+ *   type squareup.dinosaurs.Dinosaur {
+ *     target com.squareup.dino.Dinosaur using com.squareup.dino.Dinosaurs#DINO_ADAPTER;
```

Comment:
Not sure how I feel about this syntax. The nesting seems arbitrary considering you can only have a single declaration inside the block.